src/main/java/org/sagebionetworks/bridge/services/StudyConsentService.java

```diff
@@ -271,23 +296,27 @@ private String loadDocumentContent(StudyConsent consent) {
             logger.info("Finished reading consent from bucket " + consentsBucket + " storagePath " +
                     consent.getStoragePath() + " (" + content.length() + " chars) in " +
                     stopwatch.elapsed(TimeUnit.MILLISECONDS) + " ms");
-            return content;
+            // Add a signature block if this document does not contain one.
+            return appendSignatureBlockIfNeeded(content);
         } catch(IOException ioe) {
             logger.error("Failure loading storagePath: " + consent.getStoragePath());
             throw new BridgeServiceException(ioe);
         }
     }
     
-    private String sanitizeHTML(String documentContent) {
-        documentContent = Jsoup.clean(documentContent, BridgeConstants.CKEDITOR_WHITELIST);
-        Document document = Jsoup.parseBodyFragment(documentContent);
-        document.outputSettings().escapeMode(EscapeMode.xhtml)
-            .prettyPrint(false).syntax(Syntax.xml).charset("UTF-8");
-        return document.body().html();
+    private String appendSignatureBlockIfNeeded(String content) {
+        // The user can change the signature block, they can remove parts of the signature block,
+        // but if the person's name or the signing date are gone, we're assuming that they've 
+        // removed too much, and we re-append the signature block.
+        if (!content.contains("${participant.name}") || !content.contains("${participant.signing.date}")) {
+            content = content + SIGNATURE_BLOCK;
+        }
+        return content;
     }
```

Comment:
Do we have any unit tests for where if the signature block already exists, we don't add another signature block?

Reply:
Added